pkg/policy/policy_peer.go

```diff
@@ -91,3 +93,64 @@ func (rc L4RuleContext) PortProto() string {
 	port := strconv.Itoa(int(byteorder.NetworkToHost(uint16(rc.Port)).(uint16)))
 	return port + "/" + proto
 }
+
+// EntityContexts maps am entity to a L4RuleContexts
+type EntityContexts map[api.Entity]L4RuleContexts
+
+// DeepCopy returns a deep copy of EntityContexts
+func (ec EntityContexts) DeepCopy() EntityContexts {
+	cpy := make(EntityContexts)
+	for k, v := range ec {
+		cpy[k] = v.DeepCopy()
+	}
+	return cpy
+}
+
+// NewEntityContexts returns a new L4RuleContexts created.
+func NewEntityContexts() EntityContexts {
+	return EntityContexts(make(map[api.Entity]L4RuleContexts))
+}
+
+// CIDRContexts maps a CIDR to a L4RuleContexts
+type CIDRContexts map[api.CIDR]L4RuleContexts
+
+// DeepCopy returns a deep copy of EntityContexts
+func (cc CIDRContexts) DeepCopy() CIDRContexts {
+	cpy := make(CIDRContexts)
+	for k, v := range cc {
+		cpy[k] = v.DeepCopy()
+	}
+	return cpy
+}
+
+// NewCIDRContexts returns a new L4RuleContexts created.
+func NewCIDRContexts() CIDRContexts {
+	return CIDRContexts(make(map[api.CIDR]L4RuleContexts))
+}
+
+// Policy contains all policies combined by each type of policy peer.
+type Policy struct {
+	// Revision contains the revision number for this calculated policy.
+	Revision uint64
+
+	// ByNumericIdentity contains the policy to be filtered by SecurityIDs
+	ByNumericIdentity SecurityIDContexts
```

Comment:
Let's drop the `By` and just call these fields `Identity`, `Entity`, `CIDR`, ...